library/core/class.request.php

```diff
@@ -1023,7 +1027,13 @@ public function url($path = '', $withDomain = false, $ssl = null) {
                 }
             }
         }
-        $parts[] = ltrim($path, '/');
+
+        // Prevent the addition of webRoot a second time if the path was already encoded.
+        $path = ltrim($path, '/');
```

Comment:
Well chances are really slim. There is not much to do about that so if it is a deal breaker we should just drop this entirely